api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2Client.java

```diff
@@ -135,6 +143,8 @@ public void close() {
         }
         // Always close the HTTP client (can't be shared).
         httpClient.close();
+        // Cancel this future to invalidate any pending log messages
+        used.cancel(true);
```

Comment:
nit: Why not do it earlier (or in `finally`)... in case closing the HTTP client or executor throws?